spec/std/callstack_spec.cr

```diff
@@ -17,9 +17,9 @@ describe "Backtrace" do
     output = `#{tempfile.path}`
 
     # resolved file line:column
-    output.should match(/#{sample} 3:10/) # callee1
-    output.should match(/#{sample} 15:3/) # callee3
-    output.should match(/#{sample} 17:1/) # ???
+    output.should match(/callee1 at #{sample} 3:10/)
+    output.should match(/callee3 at #{sample} 15:3/)
```

Comment:
What happens is that the `__debug_line` section is correctly parsed (from the `.dwarf` file) then lines in the backtrace originating from `exception.cr` and `callstack.cr` files are skipped (they don't matter to the exception). Yet, the `__debug_info` isn't parsed correctly, and we fallback to the raw symbols (it should be `callee1` and `callee3` only.

We should be able to parse this section. macOS outputs DWARF v2 so maybe there is some noticeable differences from DWARF v4 as generated by Linux, FreeBSD or OpenBSD; maybe we must parse the `__debug_aranges` section instead?

I need access to a Mac to debug that. If someone is motivated enough to investigate, looking at the `.dwarf` Mach-O file, and checking what happens when I parse it. Or maybe send me both an executable and its associated dwarf file, so I can look deeper.